libmamba/src/core/virtual_packages.cpp

```diff
@@ -236,7 +236,52 @@ namespace mamba
             {
                 return "x86_64_v2";
             }
+#elif defined(_MSC_VER) && defined(_M_X64)
+            // We apply boolean masks to the results of the cpuid instruction
+            // to determine if the CPU supports the required instruction sets.
+            // The masks are generated by shifting 1 to the left by the bit
+            // position of the required feature.
+            //
+            // See those sources to understand bitshifts:
+            // https://docs.microsoft.com/en-us/cpp/intrinsics/cpuid-cpuidex?view=msvc-160
+            // https://en.wikipedia.org/wiki/CPUID#EAX=7,_ECX=0:_Extended_Features
+
+            int cpu_info[4];
+            int eax_value = 7;
+            int ecx_value = 0;
+            __cpuidex(cpu_info, eax_value, ecx_value);
+
+            bool avx512f = cpu_info[1] & (1 << 16);
+            bool avx512dq = cpu_info[1] & (1 << 17);
+            bool avx512cd = cpu_info[1] & (1 << 28);
+            bool avx512bw = cpu_info[1] & (1 << 30);
+            bool avx512vl = cpu_info[1] & (1 << 31);
+            if (avx512f && avx512dq && avx512cd && avx512bw && avx512vl)
```

Comment:
Nitpicking: avx512 dq/cd/bw/vl are all implemented on top of avx512f, so no need to check for this value.

Reply:
I agree, but I propose to keep the logic in line with the one for unix system as currently implemented a few lines above because it is cheap and it does not make assumption (also, I would take time to make sure that the above is correctly implemented, because more instructions seem to be present for each architecture).